src/OpenLoco/src/GameCommands/Airports/CreateAirport.cpp

```diff
@@ -25,19 +25,75 @@ namespace OpenLoco::GameCommands
     static loco_global<World::Pos2, 0x00112C792> _lastConstructedAdjoiningStationCentrePos; // Can be x = -1 for no adjoining station
 
     // 0x00490372
-    static StationManager::NearbyStation sub_490372(World::Pos3 pos, uint8_t airportObjectId, uint8_t rotation)
+    static StationManager::NearbyStation findNearbyStationAirport(World::Pos3 pos, uint8_t airportObjectId, uint8_t rotation)
     {
-        registers regs;
-        regs.eax = (pos.x & 0xFFFFU); // eax as we need to empty upper portion of eax
-        regs.cx = pos.y;
-        regs.dx = pos.z;
-        regs.bl = airportObjectId;
-        regs.bh = rotation;
-        call(0x00490372, regs);
-        StationManager::NearbyStation result{};
-        result.id = static_cast<StationId>(regs.bx);
-        result.isPhysicallyAttached = regs.eax & (1U << 31);
-        return result;
+        // This function is very similar to StationManager::findNearbyStation differences are marked
+
+        const auto companyId = getUpdatingCompanyId();
+        // Change from StationManager::findNearbyStation
+        auto* airportObj = ObjectManager::get<AirportObject>(airportObjectId);
+        const auto [minExtent, maxExtent] = airportObj->getAirportExtents(World::toTileSpace(pos), rotation);
+
+        // Check area including a 2 tile border around the airport
+        const auto tilePosA = minExtent - World::TilePos2(2, 2);
+        const auto tilePosB = maxExtent + World::TilePos2(2, 2);
+
+        auto minDistanceStation = StationId::null;
+        auto minDistance = std::numeric_limits<int16_t>::max();
+        bool isPhysicallyAttached = false;
+        for (const auto tilePos : World::getClampedRange(tilePosA, tilePosB))
+        {
+            const auto tile = World::TileManager::get(tilePos);
+            for (auto& el : tile)
+            {
+                auto* elStation = el.as<World::StationElement>();
+                if (elStation == nullptr)
+                {
+                    continue;
+                }
+                if (elStation->isGhost())
+                {
+                    continue;
+                }
+                auto* station = StationManager::get(elStation->stationId());
+                if (station->owner != companyId)
+                {
+                    continue;
+                }
+
+                const auto distance = Math::Vector::chebyshevDistance2D(World::toWorldSpace(tilePos), pos);
+                if (distance < minDistance)
+                {
+                    auto distDiffZ = std::abs(elStation->baseHeight() - pos.z);
+                    if (distDiffZ > 64)
```

Comment:
I think we should have a constant defined for this, StationManager also has a few of those, perhaps something like kMaxStationSearchDistance